client/src/partials/reports/global_transaction/global_transaction.js

```diff
@@ -151,33 +146,43 @@ angular.module('bhima.controllers')
         datef : util.sqlDate($scope.dates.from),
         datet : util.sqlDate($scope.dates.to)
       };
+      var url = '/reports/allTransactions/?source=' + $scope.model.source_id +'&enterprise_id=' + $scope.enterprise.id + '&account_id=0' + '&datef=' + util.sqlDate($scope.dates.from) + '&datet=' + util.sqlDate($scope.dates.to);
 
-      connect.fetch('/reports/allTrans/?'+JSON.stringify(qo))
+      connect.fetch(url)
       .then(function (res) {
-        if (res.length > 0) {
-          res.map(function (item) {
-            item.debit = getValue(map[util.sqlDate(item.trans_date)], item.debit, $scope.enterprise.currency_id);
-            item.credit = getValue(map[util.sqlDate(item.trans_date)], item.credit, $scope.enterprise.currency_id);
-          });
-          $scope.records = res;
-          getTotal(res);
-        } else {
-          $scope.somCredit = 0;
-          $scope.somDebit = 0;
-          $scope.records = [];
-        }
+        res.map(function (item) {
+            if($scope.enterprise.currency_id !== item.currency_id){
+              item.debit *= exchange.rate(item.debit,item.currency_id,new Date());
+              item.credit *= exchange.rate(item.debit,item.currency_id,new Date()); 
+            }
+        });
+        $scope.records = res;
+        getTotal(res);
       });
     }
 
     function getTotal(items) {
+      var sCredit = 0, 
+        sDebit = 0;
       $scope.somCredit = 0;
       $scope.somDebit = 0;
-      if (items.length > 0) {
-        items.forEach(function (item) {
-          $scope.somDebit+=item.debit;
-          $scope.somCredit+=item.credit;
-        });
-      }
+      items.forEach(function (item) {
+        if($scope.enterprise.currency_id !== item.currency_id){
+          sCredit += item.credit / exchange.rate(item.credit,item.currency_id,new Date()); 
+          sDebit += item.debit / exchange.rate(item.debit,item.currency_id,new Date());
+        } else {
+          sCredit += item.credit;
+          sDebit += item.debit;
+        }
+                
+        if($scope.enterprise.currency_id === $scope.model.c){
+          $scope.somCredit = sCredit;
```

Comment:
$scope.somCredit is updated with every iteration of the .forEach() loop.  Are you sure this is what you want to do?